sycl/include/CL/sycl/INTEL/ac_int.hpp

```diff
@@ -0,0 +1,2458 @@
+//==----------------- ac_int.hpp --- SYCL FPGA AC data-type ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Algorithmic C (tm) Datatypes                                         
+//                                                                      
+// Software Version: 3.7                                                
+//                                                                      
+// Release Date    : Wed Jun  1 13:21:52 PDT 2016                       
+// Release Type    : Production Release                                 
+// Release Build   : 3.7.0                                              
+//                                                                      
+// Copyright 2004-2016, Mentor Graphics Corporation,                    
+//                                                                      
+// All Rights Reserved.                                                 
+//                                                                      
+// This file was modified by the Intel High Level Design team to        
+// generate efficient hardware for the Intel High Level Synthesis       
+// compiler. The API remains the same as defined by Mentor Graphics     
+// in their documentation for the ac_int data-type.
+//                     
+//  Source:          ac_int.hpp
+//  Description:     fast arbitrary-length bit-accurate integer types:
+//                     - unsigned integer of length W:  ac_int<W,false>
+//                     - signed integer of length W:  ac_int<W,true>
+//  Original Author: Andres Takach, Ph.D.
+//  Modified by:     Vince Bridgers, Thor Thayer, Ajaykumar Kannan
+//
+//  Notes:
+//   - Most frequent migration issues:
+//      - need to cast to common type when using question mark operator:
+//          (a < 0) ? -a : a;  // a is ac_int<W,true>
+//        change to:
+//          (a < 0) ? -a : (ac_int<W+1,true>) a;
+//        or
+//          (a < 0) ? (ac_int<W+1,false>) -a : (ac_int<W+1,false>) a;
+//
+//      - left shift is not arithmetic ("a<<n" has same bitwidth as "a")
+//          ac_int<W+1,false> b = a << 1;  // a is ac_int<W,false>
+//        is not equivalent to b=2*a. In order to get 2*a behavior change to:
+//          ac_int<W+1,false> b = (ac_int<W+1,false>)a << 1;
+//
+//      - only static length read/write slices are supported:
+//         - read:  x.slc<4>(k) =>
+//                         returns ac_int for 4-bit slice x(4+k-1 DOWNTO k)
+//         - write: x.set_slc(k,y) = writes bits of y to x starting at index k
+
+#ifndef __ALTR_AC_INT_H
+#define __ALTR_AC_INT_H
+
+#define AC_VERSION 3
```

Comment:
Can the version be reset or some other code relies on it?

Reply:
This version number comes from upstream: 
https://github.com/hlslibs